bftengine/src/bcstatetransfer/RangeValidationTree.hpp

```diff
@@ -271,25 +321,36 @@ class RangeValidationTree {
   void setNewRoot(const RVTNodePtr& new_root);
   RVTNodePtr openForInsertion(uint64_t level) const;
   RVTNodePtr openForRemoval(uint64_t level) const;
+  uint64_t rvbIdToIndex(RVBId rvb_id) const;  // rvb_id must be valid
+  uint64_t rvbIndexToId(RVBId rvb_id) const;  // rvb_id must be valid
+
+  enum class ArrUpdateType { ADD_NODE_TO_RIGHT, CHECK_REMOVE_NODE };
+  void updateOpenRvtNodeArrays(ArrUpdateType update_type, RVTNodePtr node);
 
  protected:
-  static constexpr size_t kMaxNodesToPrint{10000};
   // vector index represents level in tree
   // level 0 represents RVB node so it would always hold 0x0
   std::array<RVTNodePtr, NodeInfo::kMaxLevels> rightmostRVTNode_;
   std::array<RVTNodePtr, NodeInfo::kMaxLevels> leftmostRVTNode_;
   std::unordered_map<uint64_t, RVTNodePtr> id_to_node_;
   RVTNodePtr root_{nullptr};
-  uint64_t max_rvb_index_{0};  // RVB index is (RVB ID / fetch range size). This is the maximal index in the tree.
-  uint64_t min_rvb_index_{0};  // RVB index is (RVB ID / fetch range size). This is the minimal index in the tree.
-
+  uint64_t min_rvb_index_{};  // RVB index is (RVB ID / fetch range size). This is the minimal index in the tree.
+  uint64_t max_rvb_index_{};  // RVB index is (RVB ID / fetch range size). This is the maximal index in the tree.
+  std::unordered_set<uint64_t> node_ids_to_erase_;
   const logging::Logger& logger_;
-  const uint32_t RVT_K{0};
-  const uint32_t fetch_range_size_{0};
-  const size_t value_size_{0};
+
+  // constants
+  static uint32_t RVT_K;
+  const uint32_t fetch_range_size_{};
+  const size_t value_size_{};
+  static constexpr size_t kMaxNodesToPrint{10000};
   static constexpr uint8_t CHECKPOINT_PERSISTENCY_VERSION{1};
   static constexpr uint8_t version_num_{CHECKPOINT_PERSISTENCY_VERSION};
   static constexpr uint64_t magic_num_{0x1122334455667788};
+  static constexpr uint8_t kDefaultRVBLeafLevel = 0;
+  static constexpr uint8_t kDefaultRVTLeafLevel = 1;
 };
 
+using LogPrintVerbosity = RangeValidationTree::LogPrintVerbosity;
```

Comment:
@mdarade , this cannot be there, since I'm not the one using it.
The idea is different than the usualy convention: I want to export it outside and prevent many other users to define it on header, so I'm doing the work for them in 1 single place - at the end of the file, only after I've defined this type. Else it won't compile.